embedded-graphics/src/fonts/mod.rs

```diff
@@ -107,200 +111,78 @@
 //! [`ArrayString`]: https://docs.rs/arrayvec/0.4.11/arrayvec/struct.ArrayString.html
 //! [`write!()`]: https://doc.rust-lang.org/nightly/std/macro.write.html
 
+mod font;
 mod font12x16;
 mod font24x32;
 mod font6x12;
 mod font6x8;
 mod font8x16;
-pub mod font_builder;
+mod text;
 
-pub use self::font12x16::Font12x16;
-pub use self::font24x32::Font24x32;
-pub use self::font6x12::Font6x12;
-pub use self::font6x8::Font6x8;
-pub use self::font8x16::Font8x16;
-use crate::geometry::Dimensions;
-use crate::pixelcolor::PixelColor;
-use crate::style::TextStyle;
+pub use font::{Font, FontIterator};
+pub use text::Text;
 
-/// Common methods for all fonts
-pub trait Font<'a, C>: Dimensions
-where
-    C: PixelColor,
-{
-    /// Render a string in the implementing font's typeface.
-    ///
-    /// Defaults to 1u8 for stroke_color and 0u8 for fill_color
-    ///
-    /// ```rust
-    /// use embedded_graphics::prelude::*;
-    /// use embedded_graphics::fonts::Font6x8;
-    /// use embedded_graphics::pixelcolor::Rgb565;
-    /// use embedded_graphics::style::TextStyle;
-    /// # use embedded_graphics::mock_display::MockDisplay as Display;
-    ///
-    /// fn main() {
-    ///     let mut disp = Display::default();
-    ///
-    ///     // Render a string with red characters
-    ///     let text = Font6x8::render_str("Hello world", TextStyle::with_text_color(Rgb565::RED));
-    ///     text.draw(&mut disp);
-    /// }
-    /// ```
-    fn render_str(chars: &'a str, style: TextStyle<C>) -> Self;
-}
+pub use font12x16::Font12x16;
+pub use font24x32::Font24x32;
+pub use font6x12::Font6x12;
+pub use font6x8::Font6x8;
+pub use font8x16::Font8x16;
 
-/// Internal macro used to implement `text_*` on fonts. Do not use directly!
-#[doc(hidden)]
+/// TODO: docs
+/// TODO: don't require font as first parameter
+/// TODO: position parameter
```

Comment:
Should `Text::new()` take a position argument, or rely on `Transform` for positioning? The primitives all take coordinates in their constructors so I understand keeping the consistency with them.

Reply:
I think `Text::new()` and the `egtext` macro should take the a position parameter, because setting the position is necessary for all texts. Another benefit of setting the position in `Text::new()` is that it is possible to create constant text objects this way.

The comment on the `egtext` macro was added because I wanted to consider different variants of adding the position parameter:
```rust
egtext!(
    "Hello!",
    Point::new(10, 20),
    font = Font24x32
);
// or with assignments for all settings
egtext!(
    text = "Hello!",
    position = Point::new(10, 20),
    font = Font24x32
);
```
But this would need changes to the primitive macros to keep them consistent:
```rust
egcircle!(
    (CIRCLE_SIZE, CIRCLE_SIZE),
    CIRCLE_SIZE as u32,
    stroke_color = Some(BinaryColor::Off),
    fill_color = Some(BinaryColor::Off)
)
// would become:
egcircle!(
    center = (CIRCLE_SIZE, CIRCLE_SIZE),
    radius = CIRCLE_SIZE as u32,
    stroke_color = Some(BinaryColor::Off),
    fill_color = Some(BinaryColor::Off)
)
```
This is a bit longer, but it's immediately clear what the values mean.

But this is totally outside the scope of this PR and I don't know if this would be easily implementable without using procedual macros. I'll simply add the position parameter without a label in the `egtext` macro for now.

Reply:
Ok cool :+1:

> This is a bit longer, but it's immediately clear what the values mean.

Very true - I like that idea. I don't think it adds too much bloat to the convenience macros. 

> But this is totally outside the scope of this PR and I don't know if this would be easily implementable without using procedual macros.

Agreed, but would be neat if it _was_ implementable.